### PR TITLE
Fix/repl tests

### PR DESCRIPTION
**Pull request by ravi-sawlani-yral**
## Description
- fixed repl tests
- added integration test for platform orchestrator

### CODE REVIEW

File: scripts/canisters/local_deploy/start_upgrades.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# !/bin/bash
+
+
+dfx build individual_user_template
+
+# Specify the path to your Wasm.gz file
+wasm=".dfx/ic/canisters/individual_user_template/individual_user_template.wasm.gz"
```
Comment (komal-rs) [inline scripts/canisters/local_deploy/start_upgrades.sh:7]: Don't we have to gzip here like we were doing previously or does dfx automatically do that. I see the wasm path is different here than the other ones. Any reason ?

File: src/lib/integration_tests/tests/platform_orchestrator/provision_subnet_orchestrator.rs [hunk 2]
```diff
@@ -87,6 +87,7 @@ fn provision_subnet_orchestrator_canister() {
     let pocket_ic = PocketIcBuilder::new()
         .with_nns_subnet()
         .with_application_subnet()
+        .with_application_subnet()
```
Comment (komal-rs) [inline src/lib/integration_tests/tests/platform_orchestrator/provision_subnet_orchestrator.rs:90]: repeated line

File: src/lib/integration_tests/tests/platform_orchestrator/provision_subnet_orchestrator.rs [hunk 3]
```diff
@@ -98,21 +99,22 @@ fn provision_subnet_orchestrator_canister() {
...
         pocket_ic.tick()
     }
-    pocket_ic.update_call(platform_canister_id, Principal::anonymous(), "upload_wasms", candid::encode_args((WasmType::SubnetOrchestratorWasm, subnet_orchestrator_canister_wasm.to_vec())).unwrap()).unwrap();
-    pocket_ic.update_call(platform_canister_id, Principal::anonymous(), "upload_wasms", candid::encode_args((WasmType::PostCacheWasm, post_cache_canister_wasm.to_vec())).unwrap()).unwrap();
+    pocket_ic.update_call(platform_canister_id, super_admin, "upload_wasms", candid::encode_args((WasmType::SubnetOrchestratorWasm, subnet_orchestrator_canister_wasm.to_vec())).unwrap()).unwrap();
+    pocket_ic.update_call(platform_canister_id, super_admin, "upload_wasms", candid::encode_args((WasmType::PostCacheWasm, post_cache_canister_wasm.to_vec())).unwrap()).unwrap();
+    pocket_ic.update_call(platform_canister_id, super_admin, "upload_wasms", candid::encode_args((WasmType::IndividualUserWasm, individual_user_template.to_vec())).unwrap()).unwrap();
     pocket_ic.add_cycles(platform_canister_id, CANISTER_INITIAL_CYCLES_FOR_SPAWNING_CANISTERS);
 
+    pocket_ic.add_cycles(platform_canister_id, 10_000_000_000_000_000);
```
Comment (komal-rs) [inline src/lib/integration_tests/tests/platform_orchestrator/provision_subnet_orchestrator.rs:117]: are we adding more additional to above line or can we change the value there itself

File: src/lib/test_utils/src/setup/test_constants/mod.rs [hunk 4]
```diff
@@ -122,25 +122,25 @@ pub fn get_canister_wasm(canister_type: KnownPrincipalType) -> Vec<u8> {
...
         KnownPrincipalType::CanisterIdPostCache => {
-            file_path.push("post_cache.wasm");
+            file_path.push("post_cache.wasm.gz");
             let mut file = File::open(&file_path)
                 .unwrap_or_else(|_| panic!("Failed to open file: {}", file_path.to_str().unwrap()));
             file.read_to_end(&mut bytes).expect("Failed to read file");
         }
         KnownPrincipalType::CanisterIdUserIndex => {
-            file_path.push("user_index.wasm");
+            file_path.push("user_index.wasm.gz");
```
Comment (komal-rs) [inline src/lib/test_utils/src/setup/test_constants/mod.rs:143]: Do the existing ic-state-machine tests work with these changes or we are not running them ? 

Ex - `when_configuration_is_updated_in_the_configuration_canister_then_it_is_propagated_to_the_other_canisters_in_the_subnet_such_as_the_user_index_canister_and_the_post_cache_canister_and_the_individual_user_canisters` is still not [ignore]

**Review by komal-sai-yral — COMMENTED** [review]
Overall looks good. Minor comments/doubts.

File: src/lib/integration_tests/tests/platform_orchestrator/provision_subnet_orchestrator.rs (hunk 2 above)
Comment (ravibazz) [reply src/lib/integration_tests/tests/platform_orchestrator/provision_subnet_orchestrator.rs:90]: This is needed. We are checking deployment on different subnet.

File: src/lib/integration_tests/tests/platform_orchestrator/provision_subnet_orchestrator.rs (hunk 3 above)
Comment (ravibazz) [reply src/lib/integration_tests/tests/platform_orchestrator/provision_subnet_orchestrator.rs:117]: Yes I will do that

**Review by komal-sai-yral — APPROVED** [review]
LGTM

**Pull request merged**